### PR TITLE
[1.21.1] Fix layering of translucent sprites being broken

### PR DESCRIPTION
**Pull request by StavWasPlayZ**
PR for issue #10114 

Read for more info.

### CODE REVIEW

**Comment by LexManos** [comment]
This seems to be a shotgun approach when a precision one is more appropriate.
Restoring the specific blend enable/disable around the blur effect. And figuring out why Mojang removed it would be the best option.
Forcing blend on all render layers could cause a ton of ill effects.

**Comment by StavWasPlayZ** [comment]
Right, I see your point.

I would *speculate* that blending was simply not necessary anymore on their end, as it would make not much sense for anything to have the need to blend with the background - considering it should be the lowest of the elements. However, in the case of overlaying—a Forge feature—it is indeed possible.

From my experience, I could not get the GUI elements to blend correctly with the blur without this patch.

> I also, notibly, tried to enable some more random features from `RenderSystem` around my elements to see if any one of them may fix it, but to no avail.

I'm uncertain of the 'ton of ill effects' you speak of (it was, evidently, present in vanilla at some point) – but this was just my surface investigation on the matter. Do feel free to dive deeper.

**Comment by LexManos** [comment]
It was present in vanilla, but also removed, for an unknown reason. Possibly because they didn't see a case where it was needed, but also could be because they found a case where it broke some rendering and we're not seeing it in your tests.
I honestly don't know and don't know where to dive deeper.
PR wise the better approach would be to re-add it to the spots vanilla removed. Thus making it not a shotgun approach.

It would be nice if you could, instead of disabeling it, make sure you set it to whatever it was before.
basically:
```
if (!isBlendEnabled()) {
    enable();
    doShit();
    disable();
  } else {
    doShit();
  }
  ```

**Comment by StavWasPlayZ** [comment]
That makes a lot of sense actually.

Just to confirm, you're saying we should add back what was initially removed in `GameRenderer#processBlurEffect`?

It's definitely better, considering we're wrapping the blur effect *itself* rather than the *whole entire* layer.

Also removes the overhead for if the background is not enabled at all. 

If you confirm, I'll get to testing and implementing later today.

**Comment by LexManos** [comment]
Yes that's what I'm saying. Make sure to make a comment in the patch noteing the issue you filed so we have a reference in the future to why it was readded.

**Comment by StavWasPlayZ** [comment]
Sure thing. Could I just get a quick guide on how I may go about modifying the `.patch` files? My regular ol' method ain't quite working here 😅

image

**Comment by LexManos** [comment]
gradlew setup
*import projects into your ide*
*edit code in forge project like a normal java project*
gradlew genPatches

**Comment by StavWasPlayZ** [comment]
Alright, I pushed and tested that it does solve the issue. Everything should work fine under that.

**Comment by StavWasPlayZ** [comment]
Yessir, done

**Comment by basdxz** [comment]
This is will may be reverted on 1.21.3, due to changes in how the render layering works causing further issues down the line.

Can you provide a reproduction case to better test this issue?

**Comment by StavWasPlayZ** [comment]
Sure - will be free by the weekend, if that's alright.

**Comment by StavWasPlayZ** [comment]
Can confirm that the issue is also resolved in Forge 53.0.8.
